ControlRoom/Controllers/SimCtl+SubCommands.swift

```diff
@@ -0,0 +1,694 @@
+        case launch(deviceId: String, appBundleId: String, waitForDebugger: Bool = false, output: Launch.Output? = nil)
+        /// Terminate an application by identifier on a device.
+        case terminate(deviceId: String, appBundleId: String)
+        /// Spawn a process by executing a given executable on a device.
+        case spawn(deviceId: String, pathToExecutable: String, options: [Spawn.Option] = [])
+        /// List available devices, device types, runtimes, or device pairs.
+        case list(filter: List.Filter? = nil, search: List.Search? = nil, flags: [List.Flag] = [])
+        /// Show the installed applications.
+        case listApps(deviceId: String, flags: [List.Flag] = [])
+        /// Trigger iCloud sync on a device.
+        case icloudSync(deviceId: String)
+        /// Sync the pasteboard content from one pasteboard to another.
+        case pbsync(source: Pasteboard.Device, destination: Pasteboard.Device, flags: [Pasteboard.Flag] = [])
+        /// Copy standard input onto the device pasteboard.
+        case pbcopy(device: Pasteboard.Device, flags: [Pasteboard.Flag] = [])
+        /// Print the contents of the device's pasteboard to standard output.
+        case pbpaste(device: Pasteboard.Device, flags: [Pasteboard.Flag] = [])
+        /// Set up a device IO operation.
+        case io(deviceId: String, operation: IO.Operation)
+        /// Collect diagnostic information and logs.
+        case diagnose(flags: [Diagnose.Flag])
+        /// enable or disable verbose logging for a device
+        case logverbose(deviceId: String?, isEnabled: Bool = false)
+        /// Set or clear status bar overrides
+        case statusBar(deviceId: String, operation: StatusBar.Operation)
+        /// Get or Set UI options
+        case ui(deviceId: String, option: UI.Option)
+        /// Send a simulated push notification
+        case push(deviceId: String, appBundleId: String? = nil, json: Push.JSON)
+        /// Grant, revoke, or reset privacy and permissi Manipulate a device's keychain
+        case privacy(deviceId: String, action: Privacy.Action, service: Privacy.Permission, appBundleId: String? = nil)
+        /// Manipulate a device's keychain
+        case keychain(deviceId: String, action: Keychain.Action)
+
+        var arguments: [String] {
+            switch self {
+            case .create(name: let name, deviceTypeId: let deviceTypeId, runtimeId: let runtimeId):
+                var arguments = ["create", name, deviceTypeId]
+                if let runtimeId = runtimeId {
+                    arguments.append(runtimeId)
+                }
+                return arguments
+            case .clone(deviceId: let deviceId, name: let name):
+                return ["clone", deviceId, name]
+            case .upgrade(deviceId: let deviceId, runtimeId: let runtimeId):
+                return ["upgrade", deviceId, runtimeId]
+            case .delete(let delete):
```

Comment:
What if instead of having this giant enum and giant switch statement, we had something like this:

```swift
struct SubCommand {
    let subcommand: String
    let arguments: [String]
}

extension SubCommand {
    static func create(name: String, deviceTypeId: String, runtimeId: String) → SubCommand { 
        return SubCommand(subcommand: "create", arguments: [name, deviceTypeId, runtimeId])
    }
    static func clone(deviceId: String, name: String) → SubCommand {
        return SubCommand(subcommand: "clone", arguments: [deviceId, name])
    }
    ...
}
```

A `static func ...` has the same syntax at the point-of-use as an enum case, but can make usage later quite a bit simpler.

Reply:
Yeah, the further I got down this list the better that started to feel in my stomach as well.  Let me give that a try!

Reply:
Alright, feels much better with the static funcs and doesn't change the api surface much at all.  The type checker does a better job with it too lol